### DISCUSSION TITLE
Vitocal 333-G mit Vitoconnect und Optolink

### DISCUSSION THREAD

**Discussion by ennimann** [philippoo66/optolink-splitter#21] [General]
Hallo,

erstmal lieben Dank für die viele Mühe und die Bereitstellung der Daten.
Aufgrund von jahrelangen Problemen mit meiner Kühlung, möchte ich gern mal ein paar Tage Daten von meiner Anlage mitloggen, um sie später in Grafen auszuwerten.
Wenn ich das richtig verstanden habe, benötige dazu einen Raspi (Raspi 2 vorhanden).
Ich nehme an, dass dieser nur eine Standard Distribution drauf benötigt und ein aktuellen Pyton.
Um jetzt z.Bsp csv Dateien zu erzeugen, müsste ich nur den Optolink an den Raspi anschließen, das Pyton Script starten und laufen lassen. Später kann ich dann die csv Dateien hernehmen und z.Bsp mit Viessdata anzeigen lassen.

Vitoconnect muss in dieser Zeit nicht laufen.
Habe ich hier einen Denkfehler oder sollte das so erstmal funktionieren?

Des weiteren wären für mich noch Daten vom NC-Box Temperatursensor wichtig, die Leistung der Sekundär- und Primärpumpe meiner WP und natürlich alle anderen Sensordaten.
Sind hier auch Ausleseadressen verfügbar, um diese dann auch auslesen zu können uns wie und wo trage ich diese ein, wenn diese in den Settings fehlen sollten?

Vielen Dank für weitere Informationen

VG
ennimann

**Comment by philippoo66** [comment]
moin @ennimann !

> benötige dazu einen Raspi (Raspi 2 vorhanden).

richtig. Raspi 2 ist ideal wegen keine Probleme mit even parity und tty0. Power reicht lang und locker (ssh Konsole, mgl keine grafische Oberfläche)

> dieser nur eine Standard Distribution drauf benötigt und ein aktuellen Pyton.

richtig

> Um jetzt z.Bsp csv Dateien zu erzeugen, müsste ich nur den Optolink an den Raspi anschließen, das Pyton Script starten und laufen lassen. Später kann ich dann die csv Dateien hernehmen und z.Bsp mit Viessdata anzeigen lassen.

du musst noch die poll_items Liste in der settings_ini.py auf die Datennpunkte deiner Lüftung anpassen. Die findest du passenderweise [hier und ff](https://community.viessmann.de/t5/Konnektivitaet/Optolink-Switch-Splitter-Vitoconnect-MQTT-amp-TCP-IP/m-p/456221#M57303) (bist du der ceeage aus den V Forum?)

du kannst die csv natürlich auch in Excel oder so angucken und Grafen draus machen. Viessdata bedürfte erstmal Anpassung der Grafik Section in der xml.

> sollte das so erstmal funktionieren?

ja :-)

>  NC-Box Temperatursensor wichtig, 

was ist eine NC Box? bisher sind nur 1-wire Temperatur-Sensoren implementiert.

> die Leistung der Sekundär- und Primärpumpe meiner WP und natürlich alle anderen Sensordaten. Sind hier auch Ausleseadressen verfügbar, 

@FransOV hatte ne ganze Menge Datenpunkte für die Vitocal zusammengesammelt. Ich find den Thread grad nicht, hab das aber kopiert, nötigenfalls lad ich dir die wieder hier hin.

> wo trage ich diese ein, wenn diese in den Settings fehlen sollten?

in die poll_items Liste. Wenn du da Hilfe brauchst, sag Bescheid. Eigentlich ist das recht einfach.

Grüsse!
Phil

**Reply by SpezialAgent** [reply]
Hallo Phil,
ich habe heute deinen Post zum Optolink-Splitter entdeckt und bin begeistert. Ich habe mir einen Raspi 3b und den 2102-Konverter bestellt.
Nun meine (erste) Frage:
Ich betreibe eine Wärmepumpe Vitocal 333-G-NC 331.B08
Kannst du mir sagen, welche Geräte-ID ich nehmen soll: die 204B oder die 204D?
Gibt es eine Liste mit Datenpunkten für diese WP?
Ich habe zusätzlich eine VitoVent 300F in Betrieb. Die Datenpunkte sind hier im Thread aufgelistet. Ich würde sie dazu packen.

P.S.
Ich bin hier Laie und habe die WP über Vitoconnect an meinen ioBroker angeschlossen. Es fehlen mir nur zuviele Datenpunkte.

Viele Grüße
Martin

**Reply by FransOv** [reply]
Allgemeine Produktbeschreibung: Vitocalxxx-G mit Vitotronic 200 (Typ WO1C) (ab 04/2012)	CU401B_G
Die Datenpunktliste ist im Excel an https://github.com/philippoo66/optolink-splitter/wiki/310-Parameter-Addresses

**Reply by SpezialAgent** [reply]
Vielen Dank. Der Raspi kommt in den nächsten Tagen und dann kann ich loslegen  .....

**Reply by philippoo66** [reply]
@SpezialAgent 

> Kannst du mir sagen, welche Geräte-ID ich nehmen soll: die 204B oder die 204D?

prinzipiell sind das ja die ersten beiden Bytes der Device ID, Adr. 0x00f8. Vielleicht liest du die einfach mal aus. Aber wenn die Werte passen, ist das wohl nicht so wichtig...

Grüsse!

**Reply by SpezialAgent** [reply]
Heute kommt der Raspi, dann versuche ich das mal mit 204B.
Gibt es zu der Vitocal eine Settings-Datei? Ich habe aus deinem Repo mal die DP_CU401B_G.TXT runtergeladen. Meine Vitocal hat nur einen Heizkreis HK1. Kann ich die anderen Datenpunkte HK2 und HK3 auskommentieren oder mach das die Software?
Gruß
Martin

**Comment by FransOv** [comment]
In Anhang eine settings_ini.py Datei  mit alle meiner Datenpunkte. Du brauchst nur die MQTT Server Addresse und userid:pwd zu ändern.
[settings_ini.py.txt](https://github.com/user-attachments/files/16402322/settings_ini.py.txt)
Die .txt ist nur wegen Github da. .py Dateien sind nicht erlaubt.

**Reply by philippoo66** [reply]
> Meine Vitocal hat nur einen Heizkreis HK1

dann kommentier die hk2 DPs in der settings_ini von Frans besser aus. wozu Sachen lesen, die nicht da sind.... optolink ist nicht die schnellste Kommunikation, und falls es keine Antwort geben sollte, ist ein Timeout stark ausbremsend....

**Comment by FransOv** [comment]
@ennimann 

> Des weiteren wären für mich noch Daten vom NC-Box Temperatursensor wichtig, 

Ich habe keine Temperatursensor im NC-Box. Benutzt du dem als Vorlauf Sensor? Ich benutze die Vorlauf Sensor vom Heizkreis. Es scheint mir als ob ein Vorlauf Sensor im NC-Box eine nervösen Reglung macht.
@philippoo66 

> was ist eine NC Box? 

https://www.viessmann.de/de/produkte/waermepumpe/nc-box.html

**Comment by ennimann** [comment]
Moin,
klasse, lieben Dank für die vielen nützlichen Infos.
Der Reihe nach.

Auf dieses Tema wurde ich von @FransOv aus dem Viessmannforum gestoßen. Danke nochmal dafür.

Die NC-Box ist die NaturalCooling Box und wird mit extra Wärmetauscher im Primärkreis der Sole und dem Rücklauf des Heiz-/Kühlkreises eingebunden.
In diesem Rücklauf sitzt ein zusätzlicher PT100 Temperatursensor der als quasi Vorlaufsensor nur bei Kühlung verwendet wird.
Eine Umstellung auf den VL Sensor der Anlage ist softwaretechnisch bei Kühlung nicht vorgesehen bzw. nicht möglich. Macht auch nach meinem Kenntnisstand keinen Sinn, da dieser quasi nur wenige cm hinter dem zusätzlichen Sensor im Verdichterraum sitzt und durch das RL Rohr auch schnell die Temperatur von der NC-Box annimmt.
Die älteren NC-Boxen hatten keinen Extrasensor drin, dafür aber eine separate Sekundärpumpe.
Da gab es in der Vergangenheit echte Verwirrung, wie die Anlagenparameter letztendlich eingestellt werden müssen, damit es funktioniert. (Parameter 7100)
Aktuell bin ich mit dem TD und dem Verkaufsshop Dresden von Viessmann intensiv dran, dass die endlich mal Ihre Software in den Griff bekommen in Bezug auf die Kühlung. (warte schon 3 Jahre auf eine laufende Version)
Um nicht ständig vor der Anlage hocken zu müssen, kommt mir dieses System hier gerade richtig, um mal belastbare Daten gegenüber Viessmann zu haben. 

Es ist also für mich essentiell wichtig, dass ich auch den PT100 der NC-Box auslesen kann, weil dieser den Startschuss für die Kühlung gibt, wenn Langzeitmittel und Kühlgrenze gerissen werden.

Ich setze erstmal den Raspi auf und schaue, dass die Anlage Daten liefert, dann gehe ich in die Einzelheiten.

Erst nochmal lieben Dank an Euch!
VG
ennimann

Neue Fragen kommen hier bestimmt bald :-)

**Comment by FransOv** [comment]
@ennimann 

> Es ist also für mich essentiell wichtig, dass ich auch den PT100 der NC-Box auslesen kann, weil dieser den Startschuss für die Kühlung gibt, wenn Langzeitmittel und Kühlgrenze gerissen werden.

Die Adresse der PT100:
WPR_Vorlauftemperatur_NatCoolingKreis~0x0119	0x0119	Div10	R	2		ecnUnit.Grad C

**Comment by ennimann** [comment]
Krass, vielen Dank!

Leider hab eich aktuell mit dem Raspi 2B absolut keine Freude, da jeder Klick ewig dauert, bis da was passiert.
Chromium und Firefox laufen auch nicht.
HAbe da jetzt echt keine Lust noch mehr Zeit reinzustecken und mit Warten zu verbringen.

Kann ich nicht einfach Python auf Windows installieren und das script so ausführen?
Will ja wirklich nur mal eine längere Session aufzeichnen, damit ich mal paar Daten habe.
Allerdings wüsste ich dann nicht, wie schon alleine die Schnittstelle konfigurieren müsste.
Zu Z80 Assembler könnte ich Euch noch was erzählen, aber das Neumodische Zeug, da bin ich einfach zu alt für.

**Comment by FransOv** [comment]
Nah, Niemals zu alt zum lernen, bin ja schon fast 79 und lerne nog täglich etwas neues.

**Comment by ennimann** [comment]
Hab mal fix FTP installiert und konnte jetzt die python Daten rüberschieben und starten.
Daten kommen an!

In den Settings habe ich die CSV aktiviert, aber ich finde nirgends eine Datei?

```
# Viessdata utils +++++++++++++++++++
write_viessdata_csv = True
viessdata_csv_path = "/home/ennimann/viessmann"
buffer_to_write = 60
dec_separator = ","
```

Habe ich noch einen Denkfehler?
Danke schon mal

**Comment by ennimann** [comment]
Läuft jetzt alles.
War wohl nur zu ungeduldig und hab jetzt mal das viessdata_util bemüht.
Lasse erstmal laufen, schaue mir die CSVs dann an und erstelle mir paar Graphen.
DAs reicht mir erstmal für den Anfang.
Noch mal vielen Dank Euch Beiden.

Ich hoffe, dass die Daten helfen, ein besseres Verständnis zu bekommen.
VG
ennimann

**Comment by FransOv** [comment]
Ich habe eine Query gemacht in die Database des VItosoft program um die Adresse zu finden. Die meiste habe ich jetzt, aber es gibt im Natural Cooling Bereich einige Unterschiede zwischen die data im Vitsosoft database und die Dokumentation der Anlage.
```
WPR_SepNCKreis_NatCooling_HystereseRaumtemperatur~0x7102 |   | 7102 Raumtemperatur-Sollwert separater Kühlkreis
WPR_SepNCKreis_NatCooling_Raumregelung~0x7103 |   | 7103 Min. Vorlauftemperatur Kühlung
WPR_SepNCKreis_NatCooling_Neigung_Raumaufschaltung~0x7104 |   | 7104 Einfluss Raumtemperaturaufschaltung Kühlkreis
WPR_SepNCKreis_NatCooling_Niveau_Kuehlkennlinie~0x7105 |   | 7105 Raumtemperaturregelung Kühlkreis
WPR_SepNCKreis_NatCooling_Neigung_Kuehlkennlinie~0x7106 |   | 7106 Rangierung Raumtemperatursensor separater Kühlkreis
WPR_SepNCKreis_NatCooling_Integralanteil_Raumregler~0x7107 |   | 7107 Hysterese Raumtemperatur Kühlkreis
WPR_SepNCKreis_NatCooling_min_Vorlauftemperatur~0x7108 |   |  
WPR_SepNCKreis_NatCooling_mit_Mischer~0x7109 |   | 7109 Freigabe Vorlauftemperatursensor Kühlkreis
WPR_SepNCKreis_NatCooling_Laufzeit_Mischer~0x710A |   |  
WPR_SepNCKreis_NatCooling_Tastband_Mischer~0x710B |   |  
WPR_SepNCKreis_NatCooling_Totband_Mischer~0x710C |   |  
WPR_SepNCKreis_NatCooling_RaumTempSens_Rangierung~0x710D |   |  
WPR3_Niveau_Kuehlkennlinie_HK1~0x7110 |   | 7110 Niveau Kühlkennlinie
WPR3_Neigung_Kuehlkennlinie_HK1~0x7111 |   | 7111 Neigung Kühlkennlinie
WPR3_Kuehlung_7116~0x7116 |   | 7116 Fernbedienung Kühlkreis
  |   | 7117 Taupunktwächter
  |   | 7118 Einschaltschwelle Kühlintegral
WPR_WO1H_Hydraulik_AktiverKuehlbetrieb~0x71FE |   | 71FE Freigabe Active Cooling

```
Die Vitosoft Daten sind vermütlich outdated und mussen mit Vorsicht benutzt werden. 
Im Anhang ein Excel mit die Datenpunkte für meine Anlage, eine Vitocal 300G. Die ist vermütlich sehr ähnlich mit die Vitocal 333G.
[Vitocal 300G Parameters.xlsx](https://github.com/user-attachments/files/16414504/Vitocal.300G.Parameters.xlsx)

**Comment by FransOv** [comment]
Ich habe mich geirrt. Auch in unserer Heizungsanlage wird die NC-Arbeit durch einen Temperatursensor in der NC-Box bestimmt.
Obwohl die minimale NC-Vorlauftemperatur auf 18 Grad eingestellt ist, kühlt die Anlage auf 17 Grad ab. Die Vorlauftemperatur für die Heizkreise bleibt jedoch bei etwa 18 Grad.
![image](https://github.com/user-attachments/assets/6ede2d76-0390-4838-8b8b-55cf3e6ba8a9)

**Comment by ennimann** [comment]
Nabend,
habe vorhin mal die unvollständige CSV von gestern gezogen und ein Diagramm erstellt.
Die Messwerte passen, die ich haben möchte und sind plausibel.
Ich lasse es jetzt mal paar Tage laufen und schaue mir die Daten dann an.
Ein paar Probleme habe ich schon gesehen, die die Kühlung behindern bzw. teils extrem verzögern.
Ich werde berichten.
VG
ennimann

**Reply by philippoo66** [reply]
freut mich dass es läuft! Allerbesten Dank auch an @FransOv für die Unterstützung!! ich bin die letzten Tage leider mit der Regelung meiner PV am kämpfen und hatte kaum Gelegenheit, hier mal reinzugucken...

Grüsse!

**Reply by philippoo66** [reply]
nochwas @ennimann 

> Hab mal fix FTP installiert und konnte jetzt die python Daten rüberschieben und starten.

du kannst auf dem Raspi Samba installieren und dann direkt aus Windoofs auf die csv zugreifen per
`\\192.168.1.123\pi_share` im Datei-Explorer... Voll praktisch, spart das FTP gewiggel. ich hab die 'Anleitung' angehängt.
[Samba Fileserver auf Raspi.pdf](https://github.com/user-attachments/files/16420175/Samba.Fileserver.auf.Raspi.pdf)

> Kann ich nicht einfach Python auf Windows installieren und das script so ausführen?

klar kannst du machen. dann lüuft nur ein Windows Rechner die ganze Zeit und frisst Strom, und die schlafen auch gern mal ein oder machen ein Update und starten neu (aber nur das Sytem, nicht das pthon skript) oder oder... Sind jedenfalss selten stabil im 'Hinntergrundeinsatz'.

Sag dem Raspi lieber dass er NICHT in die grafische Oberfläche booten soll und geh mit ssh Console drauf:
`ssh <user>@<ip-address>`
Da gewöhnt man sich super schnell dran und du brauchst die grafiche Oberfläche eigentlich nicht. Dann hat der 2er genug Resourcen um fix zu performen. Und wenn du sie doch mal wieder brauchen solltest: `sudo raspi-config` ...

**Comment by FransOv** [comment]
@ennimann Mit dass Standard Poll Interval von 30 Sekunden registriert man oft nicht wenn die Pumpen kurz laufen. 10 Sekunden ist vielleicht die bessere Wert.
Dies sind die Daten beim Start NC-Cooling mit 10 Sekunden polling interval:
<img width="482" alt="image" src="https://github.com/user-attachments/assets/094f5d5c-9934-4b12-9a3b-6cc501ad7fae">

**Comment by ennimann** [comment]
Ich danke Euch Beiden.
Das mit den 10s Polling werde ich auch einstellen. Das ist echt ne gute Idee.
Aktuell habe ich einen Monitor am Raspi und er läuft in die Grafikoberfläche. Das passt aber so ganz gut, da habe ich die Terminalfenster gleich auf einen Blick.
Das Script läuft ja nun schon eine ganze Weile ohne Probleme.

Ist schon krass, was die WP so alles treibt, wo man sich fragt, was macht die da eigentlich?
Also mal ehrlich, von Programmierung haben die da echt keine Ahnung.
Kein Versprechen halten die ein. Es gibt bis heute keine Smartlösung von Viessmann, die funktionsfähig ist. traurig.
So, nun genug gejammert.

**Comment by SpezialAgent** [comment]
Ich habe den Optosplitter heute zum Laufen gebracht. Ich habe dazu die Liste von @FransOv genommen. Wie kann ich weitere Datenpunkte dazu nehmen? Z.B. die Vitovent 300f und die NC-Box. Ich komme leider mit der Datenpunkt-Liste nicht zurecht.
Ich habe eine Vitocal 333-G-NC 331.B08 mit der Geräte-ID 204B.

**Reply by FransOv** [reply]
Vitocal 333-G ist die Liste in Tabel CU401B-G. Der ist für alle Vitocalxxx-G mit Vitotronic 200 (Typ WO1C) (ab 04/2012)

**Reply by philippoo66** [reply]
@SpezialAgent wenn Frans' Antwort deine Fragen nicht klärt, sag mal genau wo's hapert

**Reply by SpezialAgent** [reply]
Ich habe es nun verstanden (hoffentlich  ;-) ). Ich konnte erfolgreich die Datenpunkte der VitoVent hinzufügen.
Nun muss ich mich mal an die Ventile und Pumpen rantrauen.
Es wird aber langsam.
Ich hatte vor Jahren mal Vcontrold im Einsatz. Das lief aber bei mir nicht zuverlässig .

Herzlichen Dank für das tolle Stück Software und die Bauanleitung.
Gruß
Martin

**Comment by FransOv** [comment]
Fúr die NC-Box brauche ich diese Paremeter:
    ("nc_cooling", 0x0492, 1, 1, False),
    ("nc_supply_temperature", 0x0119, 2, 0.1, False),
    ("nc_supply_target_temperature", 0x1804, 2, 0.1, False),
Die Parameter der NC-Box in die Excel sind nicht alle korrekt,  betreffen vermutlich eine frühere Software Version and möglich eine andere Version der NC-Box (mit Mischer und Pumpe).

**Reply by SpezialAgent** [reply]
@FransOv 
Danke. Ich habe die DPe eingetragen. Die Werte kommen auch alle.
Gruß
Martin

**Comment by SpezialAgent** [comment]
[settings_ini.py.txt](https://github.com/user-attachments/files/17950155/settings_ini.py.txt)
Hier ist meine settings_ini.py Datei mit meinen Vitovent-Werten und einigen Statistik-Werten.
Die Statistikwerte stimmen mit den Werten der VitoConnect nicht überein. Ich habe keinen Multiplikator gefunden, so dass ich die Werte aus dem Optisplitter benutzen kann.
Vielleicht kann mit Jemand bei den Werten "LZ_" (== Laufzeit) und "Anz_" (== Anzahl / Taktung) helfen.
Vielen Dank und Gruß
Martin

**Comment by FransOv** [comment]
Laufzeit verdichter 0x5005 ist nicht die gesamte Laufzeit, hat eine maximale Wert von 1800 Sekunden. 0x0580 stimmt überein mit die Vicare App. 0x0500 gibt die Anzahl Takten. 
0x058C ist Laufzeit Active Cooling, ich dachte du hast Natural Cooling.

**Reply by SpezialAgent** [reply]
Ich habe die Punkte korrigiert. Ja, den AC muss ich rausnehmen, ich habe Natural Cooling. Ich arbeite mich so langsam in die Thematik ein. Ich habe jetzt auch die Excel verstanden.
Ich baue gerade auch die VIS in ioBroker auf die neuen Datenpunkte um.
Danke für eure Hilfe. 
Gruß
Martin
